impl/src/main/java/slack/SlackResourceImpl.java

```diff
@@ -43,11 +49,31 @@ public SlackResourceImpl(SlackConfig slackConfig) {
             .getEmail());
     }
 
+    private <T> Observable<T> callSlack(Callable<T> callable) {
+        return fromCallable(callable).subscribeOn(Schedulers.io());
+    }
+
+    @Override
+    public Observable<String> getUserId(String email) {
+        return callSlack(() -> slack.methods().usersLookupByEmail(UsersLookupByEmailRequest
```

Comment:
Wouldn't readability be improved by breaking out the request object to a separate var?

Reply:
That goes for multiple places in this file I just noticed

Reply:
Fixed